Assets/Scripts/Combat/Abilities/Confection.cs

```diff
@@ -0,0 +1,277 @@
+            {
+                SliderScript.enabled = true;
+                BakeCanvasGroup.alpha = VisibleAlpha;
+            }
+        }
+        else
+        {
+            if(canvas.name == "BrewCanvas")
+            {
+                BrewCanvasGroup.alpha = InvisibleAlpha;
+            }
+            else if(canvas.name == "BakeCanvas")
+            {
+                SliderScript.enabled = false;
+                BakeCanvasGroup.alpha = InvisibleAlpha;
+            }
+        }
+    }
+
+    new public void StartAbility()
+    {
+        base.StartAbility();
+        Debug.Log("Started Confection Ability");
+        EnableCanvas(BrewCanvas, true);
+        CurrentDamage = 0;
+    }
+
+    private void Update()
+    {
+        if(phase == Phase.Brew)
+            BrewUpdate();
+
+        if (phase == Phase.Bake)
+            BakeUpdate();
+    }
+
+    private void BrewUpdate()
+    {
+        if (timer.IsInProgress())
+        {
+            float progress = timer.GetProgress();
+
+            SweetsDropped = CookingPot.GetComponent<ItemSlot>().GetSweetsDropped();
+            RotsDropped = CookingPot.GetComponent<ItemSlot>().GetRotsDropped();
+        }
+        else
+        {
+            EndBrewPhase();
+        }
+    }
+    
+    private void BakeUpdate()
+    {
+        if (timer.IsInProgress())
+        {
+            GoodClicks = SliderScript.GetGoodClicks();
+            PerfectClicks = SliderScript.GetPerfectClicks();
+            NumberOfClicks = SliderScript.GetNumberOfClicks();
+            if(NumberOfClicks == 3)
+                timer.StopTimer();
+        }
+        else
+        {
+            //end bake
+            EnableCanvas(BakeCanvas, false);
+            phase = Phase.Inactive;
+            CalculateBakeDamage();
+            EndAbility();
+        }
+    }
+
+    private void EndBrewPhase()
+    {
+        Debug.Log($"Brew Complete with candy: {SweetsDropped}");
+        Debug.Log($"Brew Complete with rots: {RotsDropped}");
+        //Reset sweets/rots
+        DragAndDrop[] allChildren = BrewCanvas.GetComponentsInChildren<DragAndDrop>();
+        foreach (DragAndDrop child in allChildren)
+        {
+            child.ResetPosition();
+        }
+
+        EnableCanvas(BrewCanvas, false);
+        CurrentDamage = (int) CalculateBrewDamage();
+        SweetsDropped = 0;
+        RotsDropped = 0;
+        CookingPot.ResetValues();
+        Debug.Log($"Brewing Build Up Damage: {CurrentDamage}");
+
+        StartBakePhase();
+    }
+
+    protected override void EndAbility()
+    {
+        //Canvas.ForceUpdateCanvases();
+        EnableCanvas(BrewCanvas, false);
+        EnableCanvas(BakeCanvas, false);
+        Debug.Log($"Confection Damage total: {CurrentDamage}");
+
+        //Only deals damage to one enemy
+        Attack attack = new Attack(CurrentDamage);
+        TargetedCombatants[Random.Range(0, TargetedCombatants.Length)].GetComponent<Combatant>().Defend(attack);
```

Comment:
afaik `Random.Range` uses **inclusive** values so `TargetedCombatants.Length` could be chosen as a value, which would be out of range and throw an error. Should be `TargetedCombatants.Length - 1` instead.